SimG4Core/Generators/src/Generator.cc

```diff
@@ -470,17 +471,22 @@ void Generator::particleAssignDaughters(G4PrimaryParticle *g4p, HepMC::GenPartic
     // value of the code compute inside TrackWithHistory
     setGenId(g4daught, (*vpdec)->barcode());
 
-    if (verbose > 2)
-      LogDebug("SimG4CoreGenerator") << "Assigning a " << (*vpdec)->pdg_id() << " as daughter of a " << vp->pdg_id();
+    int status = (*vpdec)->status();
+    if (verbose > 1)
+      LogDebug("SimG4CoreGenerator::::particleAssignDaughters")
+          << "Assigning a " << (*vpdec)->pdg_id() << " as daughter of a " << vp->pdg_id() << " status=" << status;
+
+    bool checkStatus = fFixG4Primary ? (status > 50 && status < 100) : status > 3;
 
-    if (((*vpdec)->status() == 2 || (*vpdec)->status() > 3) && (*vpdec)->end_vertex() != nullptr) {
+    if ((status == 2 || checkStatus || (status == 23 && std::abs(vp->pdg_id()) == 1000015)) &&
+        (*vpdec)->end_vertex() != nullptr) {
```

Comment:
If `fFixG4Primary` is false, then these lines become
```
    if ((status == 2 || status > 3 || (status == 23 && std::abs(vp->pdg_id()) == 1000015)) &&
        (*vpdec)->end_vertex() != nullptr) {
```
which does not correspond to neither what is in 12_4_X before this PR, nor to the conditions that was there before previous #39440.

Therefore, if `checkStatus` is false, then neither the 12_4_X version previous to Sep 21 is restored, nor the 12_4_X version betweem Sep 21 and now. That looks odd, the default should correspond to the version used for the already started productions.

Reply:
For me it is the same thing but it is possible to change, so the code will be identical.

Reply:
Done